docs/page-objects/index.rst

```diff
@@ -0,0 +1,121 @@
+.. _page-objects:
+
+============
+Page objects
+============
+
+A page object is a code wrapper for a webpage that implements the logic to
+parse the raw webpage data into structured data.
+
+To use web-poet, :ref:`define page object classes <page-object-classes>` for
+your target websites, and :ref:`get the output item <output-item>` using a
+:ref:`web-poet framework <frameworks>`.
+
+.. _page-object-classes:
+
+Defining a page object class
+============================
+
+A page object class is a Python class that:
+
+-   Subclasses :class:`~web_poet.pages.Injectable` or is registered or decorated
+    with :class:`Injectable.register <web_poet.pages.Injectable.register>`.
```

Comment:
Some page object features are not supported with Injectable: fields, test generation (or something else about tests, I don't recall), input validation. So I wonder if we should recommend ItemPage by default, and maybe explain Injectable separately, if there is a need.

Reply:
The same might apply to from-ground-up document; it uses Injectable.

Reply:
I agree with recommending `ItemPage` here.

However, I also think this section needs to be accurate about what is a page object. Is a class inheriting directly from `Injectable` a valid page object? If so, then that’s the “minimum” base class that makes a class a page object.

Reply:
Yeah, I was thinking about the same. It seems raw Injectable doesn't have all the page object features now - no validation, no fields, tests don't always work. So, probably we should say that page object is an instance of ItemPage, and Injectable is for more advanced use cases, if some ItemPage features get in a way - does it make sense? E.g. it should be fine to define a dependency like SemanticMarkupData (which itself depends on HtpResponse) as a subclass of Injectable.

Reply:
https://github.com/scrapinghub/web-poet/pull/152/commits/ba33547fcd27ec843c522eb57934ec84a9a99609